daemon/state.go

```diff
@@ -97,12 +97,16 @@ func (d *Daemon) restoreOldEndpoints(dir string, clean bool) (*endpointRestoreSt
 			continue
 		}
 
-		ep.Mutex.Lock()
+		if lockerr := ep.LockAlive(); lockerr != nil {
```

Comment:
This can be an unconditional lock. We are not serving the API yet and using `LockAlive()` suggests that the endpoint to be restored could be manipulated.

Reply:
done